hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockPlacementPolicyRackFaultTolerant.java

```diff
@@ -237,14 +237,11 @@ public BlockPlacementStatus verifyBlockPlacement(DatanodeInfo[] locs,
       // only one rack
       return new BlockPlacementStatusDefault(1, 1, 1);
     }
-    // 1. Check that all locations are different.
-    // 2. Count locations on different racks.
-    Set<String> racks = new TreeSet<>();
-    for (DatanodeInfo dn : locs) {
-      racks.add(dn.getNetworkLocation());
-    }
-    return new BlockPlacementStatusDefault(racks.size(), numberOfReplicas,
-        clusterMap.getNumOfRacks());
+    // Count locations on different racks.
+    final long rackCount = Arrays.stream(locs)
+        .map(DatanodeInfo::getNetworkLocation).distinct().count();
+    return new BlockPlacementStatusDefault(Math.toIntExact(rackCount),
```

Comment:
You are following the same as in HDFS-14102 but I'd prefer extracting the int casting instead of doing it in the call.
A little more verbose but I would extract it.

Reply:
Removed the casting. Thank you @goiri.